src/containers/views/DataFileNew.js

```diff
@@ -35,57 +55,123 @@ export class DataFileNew extends Component {
   }
 
   routerWillLeave(nextLocation) {
-    if (this.props.datafileChanged) {
+    if (!this.state.guiView && this.props.datafileChanged) {
       return getLeaveMessage();
     }
   }
 
-  handleClickSave() {
+  toggleView() {
+    this.setState({ guiView: !this.state.guiView });
+  }
+
+  handleChange(e) {
+    const { onDataFileChanged } = this.props;
+    (e.target.value);
+    onDataFileChanged();
+  }
+
+  handleClickSave(e) {
     const { datafileChanged, putDataFile } = this.props;
+
+    // Prevent the default event from bubbling
+    preventDefault(e);
+
+    let filename;
     if (datafileChanged) {
```

Comment:
we're not sending a data object to the `put` action via the GUI, so this shouldn't matter..